nats.go

```diff
@@ -79,18 +79,21 @@ var (
 	ErrStaleConnection      = errors.New("nats: " + STALE_CONNECTION)
 )
 
-var DefaultOptions = Options{
```

Comment:
@nogoegst I am torn. Although I fully agree with you that using `DefaultOptions` as a struct may lead to errors, especially with the presence of pointers (there is already one, and others may be added in the future), making this change would break potentially lots of users (arguably, they should vendor if they are concerned about breaking changes). The worse, is that the use of `DefaultOptions` was safe until 1.2.2, where the pointer was added.

Leaving `DefaultOptions` and adding `GetDefaultOptions()` would avoid breaking users' existing code base. I was thinking that we could remove the `Dialer` from `DefaultOptions` structure, add a comment that `DefaultOptions` is deprecated and users should use `GetDefaultOptions()` instead, and make sure that nothing is ever added to `DefaultOptions` structure itself, but instead in `GetDefaultOptions()`.

Removing the `Dialer` from `DefaultOptions` is not going to break users (the ones at 1.2.2) since the `Connect()` call is using the exact same `net.Dialer` with `DefaultTimeout` if no `Dialer` is set in `Options`.

I can do all these changes myself, but I wanted to give you the opportunity to discuss or implement those changes since you have originally submitted the PR. Let me know how you want to proceed.